src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -516,8 +520,12 @@ private BoundExpression GetExtensionMemberAccess(SyntaxNode syntax, BoundExpress
                 case NamedTypeSymbol namedTypeSymbol:
                     bool wasError = false;
 
-                    return BindTypeMemberOfType(receiver, namedTypeSymbol.Name, namedTypeSymbol,
-                        syntax, right: syntax, diagnostics, ref wasError);
+                    if (!typeArgumentsOpt.IsDefault)
+                    {
+                        namedTypeSymbol = ConstructNamedTypeUnlessTypeArgumentOmitted(syntax, namedTypeSymbol, typeArgumentsSyntax, typeArgumentsOpt, diagnostics);
```

Comment:
:memo: This fixes a bug with a generic type in an extension. It's illustrated by `ExtensionMemberLookup_MatchingExtendedType_GenericType_GenericMember` amongst other tests.